### PR TITLE
#87 Other rho statistics

### PR DESCRIPTION
**Pull request by msimet**
Here's the code for the other rho statistics (rho2-rho5).  I used the definitions from the DES shear catalog paper (https://arxiv.org/abs/1507.05603, equations 3.19-3.22) although please do check that I got them right!  I used our usual "sigma" shape parameter rather than explicitly using the trace as the DES paper does, however.

There are a few whitespace fixes in here too, sorry for the messy diff.

### CODE REVIEW

File: stile/sys_tests.py [hunk 1]
```diff
@@ -692,6 +705,160 @@ def __call__(self, data, data2=None, random=None, random2=None, config=None, **k
         return self.getCF('gg', new_data, new_data2, new_random, new_random2,
                           config=config, **kwargs)
 
+class Rho2SysTest(BaseCorrelationFunctionSysTest):
+    """
+    Compute the correlation of star shapes with residual star shapes (star shapes - psf shapes).
+    """
+    short_name = 'rho2'
+    long_name = 'Rho2 statistics (Auto-correlation of star-PSF shapes)'
```
Comment (HironaoMiyatake) [inline stile/sys_tests.py]: In parenthesis, "Correlation of star shapes and star-PSF shapes".

File: stile/sys_tests.py [hunk 2]
```diff
@@ -692,6 +705,160 @@ def __call__(self, data, data2=None, random=None, random2=None, config=None, **k
...
+    objects_list = ['star PSF']
+    required_quantities = [('ra', 'dec', 'g1', 'g2', 'psf_g1', 'psf_g2', 'w')]
+
+    def __call__(self, data, data2=None, random=None, random2=None, config=None, **kwargs):
```
Comment (HironaoMiyatake) [inline stile/sys_tests.py]: rho2 is correlation between PSF shapes and star-PSF shapes, I think we should put `data['psf_g1']` to `new_data['g1']` and then put `new_data` to `self.getCF()`.

**Comment by HironaoMiyatake** [comment]
Sorry I was being slow, but I guess rho2 is wrong? Otherwise it looks good.

**Comment by msimet** [comment]
I think I fixed rho2, does it look all right to you?

**Comment by HironaoMiyatake** [comment]
Yes, looks nice! Please merge this into master.

**Pull request merged**